uamqp/async_ops/client_async.py

```diff
@@ -276,7 +279,7 @@ def __init__(
             self._keep_alive_thread = None
         if not self._session:
             return  # already closed.
-        if not self._connection.cbs:
+        if not self._using_cbs_session:
```

Comment:
Even if we're not use the CBS session for sending/receiving - do we really want to destroy it? Isn't the CBS session used to authenticate all the other sessions using the same connection?

Reply:
Same comment for sync client.

Reply:
I think this can be spitted into two questions:

Q1: Do we need to destroy these normal sessions?

The answer is: Yes.

1. If we don't destroy these normal sessions here, then there is no other place to handle these useless sessions (to free the underlying c objects). Unless we keep tracking sessions on a connection, and release all the sessions when we're shutting down a connection.

2. Consider the situation we're using SASLPlain, there would be no cbs session in this case, just normal sessions.

Q2: How the CBS session authenticates other sessions?

The answer may surprise you, when I was playing with C code, I found in eventhub use case we don't need to use cbs session to authenticate for other sessions. We just need to put the token one time at the very beginning then every following session can work - no need to put extra tokens for those new session.

Personally I think it's the generated token authenticates access to all the resources under certain eventhub namespace.